aperturedb/ImageDataCSV.py

```diff
@@ -152,31 +152,62 @@ def load_image(self, filename):
         return False, None
 
     def check_image_buffer(self, img):
-        try:
-            decoded_img = cv2.imdecode(img, cv2.IMREAD_COLOR)
-
-            # Check image is correct
-            decoded_img = decoded_img if decoded_img is not None else img
 
+        if not self.check_image:
             return True
-        except:
+
+        try:
+            # This is a full decoding of the image, expesive operation.
+            # decoded_img = cv2.imdecode(img, cv2.IMREAD_COLOR)
+            # if decoded_img is None:
+            #     logger.error("Error decoding image")
+            #     print(img[0], img[1], img[2], img[3], img[4], img[5], img[6], img[7])
+            #     return False
+
+            # Instead, we check the head of the image.
+            # Check header signature for file format:
+            # https://en.wikipedia.org/wiki/List_of_file_signatures
+
+            # NOTE: Make sure we are checking for the same values
+            #       as Image.cc on ApertureDB
+            # Ideally, we don't want this logic here because it is replicated
+            # on the server side, and we have to maintain it in both places.
+            # But because it is expensive to send the image to the server
+            # only to find out it is bad, we do it here.
+
+            # JPEG
+            if img[0] == 255 and img[1] == 216:
+                return True
+            # PNG
+            if img[0] == 137 and img[1] == 80 and img[2] == 78 and img[3] == 71:
+                return True
+            # TIF
+            if img[0] == 73 and img[1] == 73 and img[2] == 42:
+                return True
+
+            # Else, we return false and fail the batch.
             return False
+        except Exception as e:
+            print(e)
+            return False
+
+        return False  # Should never reach here
 
     def load_url(self, url):
         retries = 0
         while True:
             imgdata = requests.get(url)
             if imgdata.ok:
                 imgbuffer = np.frombuffer(imgdata.content, dtype='uint8')
-                if self.check_image and not self.check_image_buffer(imgbuffer):
-                    logger.error("IMAGE ERROR: ", url)
+                if not self.check_image_buffer(imgbuffer):
+                    logger.warning("IMAGE ERROR: ", url)
```

Comment:
This would not show the message on console. Should such an error not be shown to the user ?

Reply:
right. Not sure. Usually, I find it more useful to find out about issues after the ingestion, on the error log file generated, rather than polluting the console. At the same time, it is good to know if something is wrong immediately, but on a large ingestion, it is not really useful. 

I will change it back, we can always control the console verbosity with `LOG_CONSOLE_LEVEL=CRITICAL`